llvm/docs/SourceLevelDebugging.rst

```diff
@@ -268,6 +279,60 @@ The formal LLVM-IR signature is:
 
 See :doc:`AssignmentTracking` for more info.
 
+.. _debug_records:
+
+Debug Records
+----------------------------
+
+LLVM also has an alternative to intrinsic functions, debug records, which
+function similarly but are not instructions. The basic syntax for debug records
+is:
+
+.. code-block:: llvm
+
+  call void llvm.dbg.<type>([metadata <arg>, ]*), !dbg <DILocation> ; Intrinsic model
```

Comment:
It feels wrong to say "the syntax for debug records is"  and then the first thing we show is _not_ a debug record.

Suggestion: remove all comments, move this line down and add a comment in the middle

```
#dbg_<type>([<arg>, ]*, <DILocation>)
; Using the intrinsic model, the above is equivalent to:
call void llvm.dbg.<type>([metadata <arg>, ]*), !dbg <DILocation>
```

Reply:
`type` is a very overloaded term, and it doesn't capture what we want here.
Maybe `kind` instead?